### PR TITLE
Add Typescript support

### PR DESCRIPTION
**Pull request by mimadrid**
Let css color be highlighted inside the typescript files (.ts, .tsx)

### CODE REVIEW

**Comment by ap** [comment]
Thanks. Sorry to take so long to reply. I was just going to merge your patch but I’m confused by the comment in the file. What is it supposed to mean…?

File: after/syntax/typescript.vim [hunk 1]
```diff
@@ -0,0 +1,5 @@
+" should already be included in the typescript syntax
```
Comment (ap) [inline after/syntax/typescript.vim]: Typescript support should already be included in the `typescript` syntax…?

**Comment by mimadrid** [comment]
Sorry for not being clear, Typescript syntax is required

**Comment by ap** [comment]
Ah, it requires a third-party syntax plugin? I am guessing that is [leafgarland/typescript-vim](/leafgarland/typescript-vim)?

**Comment by mimadrid** [comment]
I think so but I don't know if we are understanding 
I want to say something like this: https://github.com/ap/vim-css-color/blob/3a49517de8d93b1443197de9820a3024f014e3a4/after/syntax/jsx.vim#L1

**Comment by ap** [comment]
Yes, I figured. That’s what confused me.

That comment talks about *other* syntaxes already being loaded, which is how the `jsx` syntax works – it causes the `javascript` and `xml` syntaxes to also be loaded, all at the same time. Does the `typescript` syntax do that?

If yes, then that’s not what the comment says. It says that the `typescript` should… already have the `typescript` syntax loaded at the same time. That makes no sense.

**Comment by ap** [comment]
Namely, tracing back what my own comment means (and actually realising it’s too short and cannot be understood…), it’s about the fact that the `init` call has an empty highlight group list (the `''` parameter). Which is because, as I mentioned, other syntaxes will have already been loaded at that point. The only thing the `jsx` support does is to turn on the support for CSS function calls (the `'css'` parameter), whereas the `xml` and `javascript` syntaxes only turn on RGB code parsing (`'hex'`).

**Comment by ap** [comment]
I’ve pushed this as 7ecfc4810f34dbcd5d22a14013f977a9aa699312. Thank you for the patch!

**Pull request closed without merging**